storage/objects/main.go

```diff
@@ -253,6 +253,24 @@ func writeEncryptedObject(client *storage.Client, bucket, object string, secretK
 	return nil
 }
 
+func writeWithKMSKey(client *storage.Client, bucket, object string, keyName string) error {
```

Comment:
Would you mind adding function comments for this and `writeEncryptedObject`? At first glance they seem the same (they aren't, but it's easy to miss how the encryption differs).